image-pipeline/expression-scatter-plots.js

```diff
@@ -134,59 +135,71 @@ async function makeExpressionScatterPlotImage(gene, page, preamble) {
   // Ideally there would be a GCS client library equivalent of those commands,
   // but brief research found none.
   const toFilePath = `cache/expression_scatter/images/${debugNonce}${webpFileName}`
-  uploadToBucket(imagePath, toFilePath, preamble)
+  uploadToBucket(imagePath, toFilePath, context)
 
   return
 }
 
 /** Fetch JSON data for gene expression scatter plot, before loading page */
 async function prefetchExpressionData(gene, context) {
-  const { accession, preamble, origin, fetchOrigin } = context
+  const { accession, preamble, origin, fetchOrigin, cluster } = context
 
   const extension = values['json-dir'] && initExpressionResponse ? '.gz' : ''
   const jsonPath = `${jsonFpStem}${gene}.json${extension}`
 
-  print(`Prefetching JSON for ${gene}`, preamble)
+  print(`Prefetching JSON for ${gene}`, context)
 
-  let isCopyOnFilesystem = true
-  try {
-    await access(jsonPath)
-  } catch {
-    isCopyOnFilesystem = false
-  }
-
-  if (isCopyOnFilesystem && initExpressionResponse) {
-    // Don't process with fetch if expression was already prefetched
-    print(`Using local expression data for ${gene}`, preamble)
-    return
-  }
-
-  // Configure URLs
-  const apiStem = `${fetchOrigin}/single_cell/api/v1`
-  const allFields = 'coordinates%2Ccells%2Cannotation%2Cexpression'
-  const params = `fields=${allFields}&gene=${gene}&subsample=all&isImagePipeline=true`
-  const url = `${apiStem}/studies/${accession}/clusters/_default?${params}`
-
-  // Fetch data
-  const response = await fetch(url)
-  const json = await response.json()
-
-  if (Object.keys(coordinates).length === 0) {
-    // Cache `coordinates` and `annotations` fields; this is done only once
-    coordinates.x = json.data.x
-    coordinates.y = json.data.y
-    if ('z' in json.data) {
-      coordinates.z = json.data.z
+  // Commented-out code is intended.  Use after incorporating `gcloud storage`.
+  //
+  // let isCopyOnFilesystem = true
+  // try {
+  //   await access(jsonPath)
+  // } catch {
+  //   isCopyOnFilesystem = false
+  // }
+  // if (isCopyOnFilesystem && initExpressionResponse) {
+  //   // Don't process with fetch if expression was already prefetched
+  //   print(`Using local expression data for ${gene} at ${jsonPath}`, context)
+  //   return
+  // }
+
+  let jsonString
+  if (!initExpressionResponse) {
+    // Configure URLs
+    const apiStem = `${fetchOrigin}/single_cell/api/v1`
+    const allFields = 'coordinates%2Ccells%2Cannotation%2Cexpression'
+    const params = `fields=${allFields}&gene=${gene}&subsample=all&isImagePipeline=true`
+    const url = `${apiStem}/studies/${accession}/clusters/_default?${params}`
+
+    // Fetch data
+    const response = await fetch(url)
```

Comment:
Should these be wrapped in some sort of error catching?
Can we assume the response will always be json-able?

Reply:
It ought to be safe to assume this will always be parseable as JSON.  If it's not, the top-level `catch` around `run()` will log the error's stack trace to the bucket, then exit.